stats/counter32.go

```diff
@@ -0,0 +1,36 @@
+package stats
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+type Counter32 struct {
+	val uint32
+}
+
+func NewCounter32(name string) *Counter32 {
+	return registry.add(name, &Counter32{}).(*Counter32)
+}
+
+func (c *Counter32) SetUint32(val uint32) {
+	atomic.StoreUint32(&c.val, val)
+}
+
+func (c *Counter32) Inc() {
+	atomic.AddUint32(&c.val, 1)
+}
+
+func (c *Counter32) Add(val int) {
+	c.AddUint32(uint32(val))
+}
+
+func (c *Counter32) AddUint32(val uint32) {
+	atomic.AddUint32(&c.val, val)
```

Comment:
yeah it'll just wrap. in fact there's 2 pressure metrics which report a counter of nanoseconds, which wraps pretty frequently. (in some tests i was loading 200k points per second and it wrapped every 20 seconds!)
graphite's `perSecond()` will introduce a null point at every wrapping point, so you need something like 983013959a5b4865e7b9baf05c5e266702a2db02 . i think this is an acceptable tradeoff.